ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java

```diff
@@ -160,14 +160,16 @@ public boolean addTransactionHash(final Hash transactionHash) {
 
   public ValidationResult<TransactionInvalidReason> addLocalTransaction(
       final Transaction transaction) {
-    final Wei transactionGasPrice = minTransactionGasPrice(transaction);
-    if (transactionGasPrice.compareTo(minTransactionGasPrice) < 0) {
-      return ValidationResult.invalid(TransactionInvalidReason.GAS_PRICE_TOO_LOW);
-    }
-
-    if (!configuration.getTxFeeCap().isZero()
-        && transactionGasPrice.compareTo(configuration.getTxFeeCap()) > 0) {
-      return ValidationResult.invalid(TransactionInvalidReason.TX_FEECAP_EXCEEDED);
+    if (transaction.isFrontierTransaction()
+        && (!ExperimentalEIPs.eip1559Enabled || this.eip1559.isEmpty())) {
+      final Wei transactionGasPrice = minTransactionGasPrice(transaction);
+      if (transactionGasPrice.compareTo(minTransactionGasPrice) < 0) {
+        return ValidationResult.invalid(TransactionInvalidReason.GAS_PRICE_TOO_LOW);
```

Comment:
We can reject local frontier transactions whose gas price is too low? Why do we do that?

Reply:
I thought we want to be very accepting of local transactions. For example, miners submit payout transactions to their own node with a very low gas price.

Reply:
Well, the behaviour of frontier transactions become the same as it was before. Those use cases are still covered by the `--min-gas-price` endpoint. Since the actual price of frontier transactions is fixed we can reject those if the price is below the configured `--min-gas-price`

Reply:
Ah ok. Then it's not an issue with the diff of this PR but something that's worth discussing about the existing behavior.